homework7 7-12-2021

SargisV opened the pull request.

SargisV committed:
homework7 7-12-2021

sirak92 commented on homeworks/sargis_vardanyan/homework7/Calculator.java:
```diff
@@ -0,0 +1,61 @@
+				}
+				return result;
+			case '/':
+				if(var[0] == 0) {
+					System.out.println("Wrong parameter input: division by 0");
+					return 0;
+				} else {
```
why you separate the first parameter? 
the 0-th element is numerator (համարիչ) and can be 0

SargisV replied:
It should be var[2] to var[i]

sirak92 commented on homeworks/sargis_vardanyan/homework7/Calculator.java:
```diff
@@ -0,0 +1,61 @@
+				} else {
+					result = var[0] * var[0];
+					for(double i: var) {
+						result /= i;
```
you have to check 0ness here

sirak92 commented on homeworks/sargis_vardanyan/homework7/DigitsToWords.java:
```diff
@@ -0,0 +1,35 @@
+	}
+	
+	
+	static void digitsToWords(int number) {
```
you solved different problem, please reread the problem statement in the slack channel

sirak92 commented on homeworks/sargis_vardanyan/homework7/PowerOfNumber.java:
```diff
@@ -0,0 +1,44 @@
+	}
+	
+	static int power(int x, int n) {
+		int pow = 1;
+		
+		if (n > 0) {
+			pow = x * power(x, n -1);
+			return pow;
+		} else if(n == 0) {
+			return pow;
+		}
+		
+		return pow;
```
it solved the problem, but could you simplify it?

SargisV replied:
Just using else { } , no need of else if condition?

sirak92 replied:
it's also a simplification but it can be simplified further.
hint: avoid indentation as much as possible

SargisV committed:
For new homework

Merge branch 'main' of github.com:sirak92/java-intro-2021 into sargis-vardanyan-homework

SargisV committed:
homework8 9-12-2021
